tests/lib/rules/no-unused-vars.js

```diff
@@ -1064,6 +1064,15 @@ ruleTester.run("no-unused-vars", rule, {
             parserOptions: { ecmaVersion: 2015 },
             errors: [{ ...assignedError("x"), line: 1, column: 23 }]
         },
+
+        // https://github.com/eslint/eslint/issues/14866
+        {
+            code: `let z = 0;
+            z = z + 1, z = 2;
+            `,
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [{ ...assignedError("z"), line: 2, column: 24 }]
+        },
```

Comment:
can you also add tests for the following cases - 

```
let z = 0;
z = z+1, z=2;
z=3;

let z = 0;
z = z+1, z=2;
z=z+3;

let x = 0; 0, x = x+1;

let x = 0; x = x+1, 0;

let x = 0; foo = ((0, x = x + 1), 0);

let x = 0; foo = (x = x+1, 0);

let x = 0; 0, (1, x=x+1);

```

Reply:
Done